tslearn/metrics/frechet.py

```diff
@@ -0,0 +1,685 @@
+            constraint, a `RuntimeWarning` is raised and no global constraint is
+            used.
+
+        itakura_max_slope : float or None (default: None)
+            Maximum slope for the Itakura parallelogram constraint.
+            If None and `global_constraint` is set to "itakura", a maximum slope
+            of 2. is used.
+            If both `sakoe_chiba_radius` and `itakura_max_slope` are set,
+            `global_constraint` is used to infer which constraint to use among the
+            two. In this case, if `global_constraint` corresponds to no global
+            constraint, a `RuntimeWarning` is raised and no global constraint is
+            used.
+
+        be : Backend object or string or None
+            Backend. If `be` is an instance of the class `NumPyBackend` or the string `"numpy"`,
+            the NumPy backend is used.
+            If `be` is an instance of the class `PyTorchBackend` or the string `"pytorch"`,
+            the PyTorch backend is used.
+            If `be` is `None`, the backend is determined by the input arrays.
+            See our :ref:`dedicated user-guide page <backend>` for more information.
+
+        Returns
+        -------
+        float
+            Similarity score
+
+        Examples
+        --------
+        >>> float(frechet([1, 2, 3], [1., 2., 2., 3.]))
+        0.0
+        >>> float(frechet([1, 2, 3], [1., 2., 2., 3., 4.]))
+        1.0
+
+        The PyTorch backend can be used to compute gradients:
+
+        >>> import torch
+        >>> s1 = torch.tensor([[1.0], [2.0], [3.0]], requires_grad=True)
+        >>> s2 = torch.tensor([[3.0], [4.0], [-3.0]])
+        >>> sim = frechet(s1, s2, be="pytorch")
+        >>> print(sim)
+        tensor(6., grad_fn=<SelectBackward0>)
+        >>> sim.backward()
+        >>> print(s1.grad)
+        tensor([[0.],
+                [0.],
+                [1.]])
+
+        >>> s1_2d = torch.tensor([[1.0, 1.0], [2.0, 2.0], [3.0, 3.0]], requires_grad=True)
+        >>> s2_2d = torch.tensor([[3.0, 3.0], [4.0, 4.0], [-3.0, -3.0]])
+        >>> sim = frechet(s1_2d, s2_2d, be="pytorch")
+        >>> print(sim)
+        tensor(8.4853, grad_fn=<SelectBackward0>)
+        >>> sim.backward()
+        >>> print(s1_2d.grad)
+        tensor([[0.0000, 0.0000],
+                [0.0000, 0.0000],
+                [0.7071, 0.7071]])
+
+        See Also
+        --------
+        frechet_path : Get both the matching path and the similarity score
+        frechet_path_from_metric : Compute similarity score and path
+            using a user-defined distance metric
+        cdist_frechet : Cross similarity matrix between time series datasets
+
+        References
+        ----------
+        .. [1] H. Sakoe, S. Chiba, "Dynamic programming algorithm optimization for
+           spoken word recognition," IEEE Transactions on Acoustics, Speech and
+           Signal Processing, vol. 26(1), pp. 43--49, 1978.
+
+    """  # noqa: E501
+    return _frechet(
```

Comment:
One slight problem with this is that we compute the path even if the used does not want it, which results in degraded performance. Shouldn't we avoid it?